src/main/java/gameframework/gui/GameWindow.java

```diff
@@ -15,20 +15,29 @@
 
 	protected final Frame frame;
 	protected GameCanvas gameCanvas;
-	protected final GameStatusBar statusBar = new GameStatusBar();
+	protected GameStatusBar statusBar = new GameStatusBar();
 	
 	public GameWindow(String gameName, GameCanvas gameCanvas, GameData data) {
-		this(gameName, gameCanvas, data.getConfiguration(),
+		this(gameName, gameCanvas, data.getConfiguration(), new GameStatusBar(),
+				new GameStatusBarElement<>("Score:", data.getScore()),
+				new GameStatusBarElement<>("Life:", data.getLife()));
+	}
+
+	public GameWindow(String gameName, GameCanvas gameCanvas, GameData data, GameStatusBar statusBar) {
+		this(gameName, gameCanvas, data.getConfiguration(), statusBar,
 				new GameStatusBarElement<>("Score:", data.getScore()),
 				new GameStatusBarElement<>("Life:", data.getLife()));
 	}
 
 	public GameWindow(String gameName, GameCanvas gameCanvas,
-			GameConfiguration configuration,
+			GameConfiguration configuration, GameStatusBar statusBar,
 			GameStatusBarElement<?>... elementsStatusBar) {
 		if (gameCanvas == null) {
 			throw new IllegalArgumentException("gameCanvas is null");
 		}
+		if (statusBar == null) {
+			this.statusBar = new GameStatusBar();
```

Comment:
`statusBar` already has an initializer along with its declaration; what you're trying to do here is simply to not overwrite it with `null`…

For code clarity I'd prefer if you refactored the constructors into a delegation chain, from the most implicit ones to a single canonical constructor that expects all parameters to be explicitly specified.